python/cudf/cudf/utils/utils.py

```diff
@@ -168,7 +168,7 @@ def pyarrow_buffer_to_cudf_buffer(arrow_buf, mask_size=0):
             dbuf = rmm.DeviceBuffer(size=mask_size)
             dbuf.copy_from_host(np.asarray(arrow_buf).view("u1"))
             return Buffer(dbuf)
-        return Buffer(arrow_buf)
+        return Buffer(np.asarray(arrow_buf).view("|u1"))
```

Comment:
Why can we not create a Buffer from an `arrow_buf` directly?

Reply:
Because of the path this case takes through the buffer constructor, I think this is why I had the cast in there at first. This hits the last case and gets converted to a `memoryview` before calling `np.asarray` on that memoryview. This results in an array with the wrong dtype and then it errors at the actual typestr check.